### PR TITLE
docs: document  ignore  in FAQ

### LINKED ISSUE #955: Document "ignore" and its limitations
.. and possibly change it to "ignored", or allow that as a synonym.

### PR DESCRIPTION
**Pull request by dholbach**
Document `ignore` annotation in the FAQ, take into account #1211 and #749.

Closes #955.

### CODE REVIEW

File: site/faq.md [hunk 1]
```diff
@@ -112,3 +112,30 @@ setting. Flux will read this value and parse the Kubernetes secret.
...
+```yaml
+    flux.weave.works/ignore: true
+```
+
+A live example of this can be seen
+[here](https://github.com/stefanprodan/openfaas-flux/blob/master/secrets/openfaas-token.yaml).
+
+Please note that [flux#749](https://github.com/weaveworks/flux/issues/749)
+limits this currently to resources of the type `namespace`, `service`
+and `deployment`.
```
Comment (rade) [inline site/faq.md]: This isn't true; see #749, which is referenced in the above.

File: site/faq.md [hunk 2]
```diff
@@ -112,3 +112,30 @@ setting. Flux will read this value and parse the Kubernetes secret.
...
+
+To stop ignoring deployments, you simply remove the annotation from
+the manifests in git.
```
Comment (rade) [inline site/faq.md]: not just 'deployments'.

File: site/faq.md (hunk 1 above)
Comment (dholbach) [reply site/faq.md]: Yes, I saw that. I was waiting for confirmation of @squaremo on https://github.com/weaveworks/flux/pull/1210#pullrequestreview-135073547

File: site/faq.md [hunk 3]
```diff
@@ -300,3 +300,32 @@ your case.
 You will need to use the (experimental) command-line flag
 `--k8s-namespace-whitelist` to enumerate the namespaces that Flux
 attempts to scan for workloads.
+
+### Can I temporarily make flux ignore a deployment?
```
Comment (squaremo) [inline site/faq.md]: If you put it in the manifest file, it'll work for any resource.

File: site/faq.md [hunk 4]
```diff
@@ -300,3 +300,32 @@ your case.
...
+
+A live example of this can be seen
+[here](https://github.com/stefanprodan/openfaas-flux/blob/master/secrets/openfaas-token.yaml).
+
+Please note that not all resources are tracked by flux at the moment,
+we are currently limited to resources of the type `namespace` and the
+set of controllers in
+[resourcekinds.go](https://github.com/weaveworks/flux/blob/master/cluster/kubernetes/resourcekinds.go),
+namely `deployment`, `daemonset`, `cronjob`, `statefulset` and `deployment`).
+
```
Comment (squaremo) [inline site/faq.md]: That's not accurate as stated. Well, depending on what's meant by "tracked".

The full story is this: flux looks at the files and the running resources when deciding whether what to apply. But it gets the running resources by exporting them from the cluster, and that only returns the kinds of resource mentioned above. So, annotating a running resource only works if it's one of those kinds; putting the annotation in the file always works.

File: site/faq.md [hunk 5]
```diff
@@ -300,3 +300,32 @@ your case.
...
+To stop ignoring these annotated resources, you simply remove the
+annotation from the manifests in git.
+
+If the annotations are carried in the cluster as well (this should
+not happen by default), you need to remove them from the cluster as well,
```
Comment (squaremo) [inline site/faq.md]: Annotating the resource in the cluster is an intended mode of use (it can be a bit easier than committing a change to git).

**Pull request merged**